.github/workflows/update-completed-sprint-on-issue-closed.yml

```diff
@@ -18,114 +18,15 @@ jobs:
-                              name
-                            }
-                          }
-                        }
-                      }
-                    }
-                  }
-                }
-              }
-            `);
-
-            const resource = data.resource;
-            if (!data.resource) {
-              throw new Error(`Issue/PR not found: ${url}`);
-            }
-
-            const projects = data.resource.projectsV2.nodes;
-            const projectItems = data.resource.projectItems.nodes?.filter(item =>
-              item.project.id === project.id
-            );
-
-            if (projects.length === 0 || projectItems.length === 0) {
-              console.log(`Item ${url} is not a member of the project: ${project_name}`);
-              return
-            }
-            if (projects.length > 1 || projectItems.length > 1) {
-              console.log(`Item ${url} is a member of multiple projects named: ${project_name}`);
-              return
-            }
-
-            const project = projects[0];
-            const projectItem = projectItems[0];
-
-            if (!project.field?.id) {
-              console.log(`${inputs.field_name} field not found on this item`);
-              return;
-            }
-
-            const sprints = project.field.options.filter(a =>
-              a.name.match(/sprint\s+(\d+)/i)
-            ).sort((a, b) => {
-              let amatch = Number(a.name.match(/sprint\s+(\d+)/i)[1]);
-              let bmatch = Number(b.name.match(/sprint\s+(\d+)/i)[1]);
-              if (amatch < bmatch) return -1;
-              if (amatch > bmatch) return 1;
-              return 0;
-            }).reverse();
-            if (sprints.length === 0) throw new Error("No last sprints found");
-
-            return {
-              item_id: projectItem.id,
-              project_id: project.id,
-              field_id: project.field.id,
-              sprint_id: sprints[0].id,
-            };
-
+          node-version: 16
+      - run: npm ci
       - name: Update the completed sprint
+        id: update_completed_sprint
         uses: actions/github-script@v5
         with:
           script: |
-            const {
-              item_id,
-              project_id,
-              field_id,
-              sprint_id,
-            } = ${{ steps.get_latest_completed_sprint.outputs }};
-
-            await github.graphql(`
-              mutation {
-                updateProjectV2ItemFieldValue(input: {
-                  projectId: "${project_id}",
-                  itemId: "${item_id}",
-                  fieldId: "${field_id}",
-                  value: {
-                    singleSelectOptionId: "${sprint_id}"
-                  }
-                }) {
-                  clientMutationId
-                }
-              }
-            `);
+            const script = require('${{ github.workspace }}/update-completed-sprint-on-issue-closed/dist/index.js');
+            await script();
```

Comment:
To use the GitHub token secret in this script as an env var, I think you need to pass it to the step. I can merge it as is though as I'm not sure. 

```yaml
env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```